src/PointwiseFunctions/GeneralRelativity/Surfaces/Tags.hpp

```diff
@@ -0,0 +1,545 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <deque>
+#include <string>
+#include <utility>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataBox/TagName.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/Strahlkorper.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/StrahlkorperFunctions.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/TagsDeclarations.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/TagsTypeAliases.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Surfaces/AreaElement.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Surfaces/EuclidianAreaElement.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Surfaces/EuclidianSurfaceIntegralOfVector.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Surfaces/ExtrinsicCurvature.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Surfaces/GradUnitNormalOneForm.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Surfaces/Mass.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Surfaces/RicciScalar.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Surfaces/Spin.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Surfaces/SurfaceIntegralOfScalar.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Surfaces/UnitNormalOneForm.hpp"
+#include "PointwiseFunctions/GeneralRelativity/TagsDeclarations.hpp"
+#include "Utilities/ForceInline.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \ingroup SurfacesGroup
+/// Holds tags and ComputeItems associated with a `::Strahlkorper`.
+namespace StrahlkorperTags {
+/// Tag for holding the previously-found values of a Strahlkorper,
+/// which are saved for extrapolation for future initial guesses
+/// and for computing the time deriv of a Strahlkorper.
+template <typename Frame>
+struct PreviousStrahlkorpers : db::SimpleTag {
+  using type = std::deque<std::pair<double, ::Strahlkorper<Frame>>>;
+};
+
+/// @{
+/// Tag to compute the time derivative of the coefficients of a Strahlkorper
+/// from a number of previous Strahlkorpers.
+template <typename Frame>
+struct TimeDerivStrahlkorper : db::SimpleTag {
+  using type = ::Strahlkorper<Frame>;
+};
+
+template <typename Frame>
+struct TimeDerivStrahlkorperCompute : db::ComputeTag,
```

Comment:
Did you intend to move this to NumericalAlgorithms/SphericalHarmonics where you also moved the time derivative function? I think the function and the tags would be best placed in the same file, perhaps `NumericalAlgorithms/SphericalHarmonics/TimeDerivStrahlkorper.*pp`.